sys/kern/src/arch/rv64/mtimer.rs

```diff
@@ -29,16 +29,28 @@ const MTIME: u64 = 0x0200_BFF8;
 //
```

Comment:
Rather than adding cfg_if checks in mtimer.rs, I think we should add a separate `stimer.rs` and including either mtimer.rs or stimer.rs conditionally. There isn't going to be any sharing between these two files except for function names.